tests/test_bundle.py

```diff
@@ -2425,3 +2425,26 @@ def test_make_manifest_bundle():
         bundle_json = json.loads(tar.extractfile("manifest.json").read().decode("utf-8"))
         assert manifest["metadata"] == bundle_json["metadata"]
         assert manifest["files"].keys() == bundle_json["files"].keys()
+
+
+def test_make_bundle_empty_manifest():
+    manifest = {}
+    with pytest.raises(TypeError):
+        make_manifest_bundle(manifest)
+
+
+def test_make_bundle_missing_file_in_manifest():
+    manifest = {
+        "version": 1,
+        "locale": "en_US.UTF-8",
+        "metadata": {"appmode": "python-shiny", "entrypoint": "app5"},
+        "python": {
+            "version": "3.8.12",
+            "package_manager": {"name": "pip", "version": "23.0.1", "package_file": "requirements.txt"},
+        },
+        "files": {
+            "requirements1.txt": {"checksum": "c82f1a9894e5510b2f0c16fa63aaa004"},
+        },
+    }
+    with pytest.raises(TypeError):
```

Comment:
I would have expected this to raise a different error, why `TypeError`?

Reply:
Existing code path has no custom exceptions, so the error we are getting back is from this file handler: https://github.com/rstudio/rsconnect-python/blob/51eb48b9652ca7adede90af89ed351c58234c2a9/rsconnect/bundle.py#L714-L726

Reply:
`make_manifest_bundle` is expecting a file path to the manifest, so you'll need to put your manifest into a file. A subdirectory in the `testdata` directory is the usual place for things like that.

Reply:
This is resolved with [79d98b5](https://github.com/rstudio/rsconnect-python/pull/395/commits/79d98b5e5dbc493833eab62635b2932ec9eb8762) and [a7c46d9](https://github.com/rstudio/rsconnect-python/pull/395/commits/a7c46d9ab3f0172d53d9f607a009bd59e66beeef).